crates/symbolicator/src/metrics.rs

```diff
@@ -6,8 +6,11 @@ use std::sync::Arc;
 use cadence::{Metric, MetricBuilder, StatsdClient, UdpMetricSink};
 use parking_lot::RwLock;
 
+// Type aliases to make the definition of [`MetricsClient`] more readable.
 type Hostname = String;
 type HostnameTag = String;
+type Environment = String;
+type EnvironmentTag = String;
```

Comment:
apologies if this ends up sounding being pretty pedantic.

from a readability standpoint, i get why this exists but i'm a little torn about it, because i'm left wondering a few things as a result of this:

- can any string be one of these four types? are there any restrictions on the contents of these strings at all?
- should you be able to use a `Hostname` as an `Environment` or `HostnameTag` or `EnvironmentTag`? you're not using a newtype here, but one of the nice perks of wrapping strings in newtypes is that it's hard to mix up different strings that represent or mean different things, ie you can't accidentally use an `Environment` as a `Hostname`.

i'm hesitant about the way we're softly baking these into the type system if it doesn't take advantage of the checking that the system provides.

Reply:
I've replaced the two dedicated tag/value pairs on `MetricsClient` with a `BTreeMap`. I believe that makes the everything both clearer and more robust.